examples/kubernetes/gluster-virtblock/csi-deployment.yaml

```diff
@@ -0,0 +1,316 @@
+            - name: pods-mount-dir
+              mountPath: /var/lib/kubelet/pods
+              mountPropagation: "Bidirectional"
+      volumes:
+        - name: plugin-dir
+          hostPath:
+            path: /var/lib/kubelet/plugins_registry/org.gluster.glustervirtblock
+            type: DirectoryOrCreate
+        - name: registration-dir
+          hostPath:
+            path: /var/lib/kubelet/plugins_registry/
+            type: Directory
+        - name: pods-mount-dir
+          hostPath:
+            path: /var/lib/kubelet/pods
+            type: Directory
+
+---
+kind: Service
+apiVersion: v1
+metadata:
+  name: csi-provisioner-glustervirtblockplugin
+  labels:
+    app: csi-provisioner-glustervirtblockplugin
+spec:
+  selector:
+    app: csi-provisioner-glustervirtblockplugin
+  ports:
+    - name: dummy
+      port: 12345
+
+---
+kind: StatefulSet
+apiVersion: apps/v1beta1
+metadata:
+  name: "csi-provisioner-glustervirtblockplugin"
+spec:
+  serviceName: "csi-provisioner-glustervirtblockplugin"
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: csi-provisioner-glustervirtblockplugin
+    spec:
+      serviceAccount: csi-provisioner
+      containers:
+        - name: csi-provisioner
+          image: quay.io/k8scsi/csi-provisioner:v1.0.1
+          args:
+            - "--provisioner=org.gluster.glustervirtblock"
+            - "--csi-address=$(ADDRESS)"
+          env:
+            - name: ADDRESS
+              value: /var/lib/csi/sockets/pluginproxy/csi.sock
+          volumeMounts:
+            - name: socket-dir
+              mountPath: /var/lib/csi/sockets/pluginproxy/
+          imagePullPolicy: "IfNotPresent"
+        - name: csi-cluster-driver-registrar
+          image: quay.io/k8scsi/csi-cluster-driver-registrar:v1.0.1
+          args:
+            - "--v=5"
+            - "--pod-info-mount-version=\"v1\""
+            - "--csi-address=$(ADDRESS)"
+          env:
+            - name: ADDRESS
+              value: /var/lib/csi/sockets/pluginproxy/csi.sock
+          volumeMounts:
+            - name: socket-dir
+              mountPath: /var/lib/csi/sockets/pluginproxy/
+        - name: glustervirtblock
+          securityContext:
+            privileged: true
+            capabilities:
+              add: ["CAP_MKNOD"]
+              add: ["CAP_SYS_ADMIN"]
+              add: ["SYS_ADMIN"]
+            allowPrivilegeEscalation: true
+          image: docker.io/gluster/glustervirtblock-csi-driver
+          args:
+            - "--nodeid=$(NODE_ID)"
+            - "--v=5"
+            - "--endpoint=$(CSI_ENDPOINT)"
+            - "--resturl=$(REST_URL)"
+            - "--resttimeout=120"
+          env:
+            - name: NODE_ID
+              valueFrom:
+                fieldRef:
+                  fieldPath: spec.nodeName
+            - name: CSI_ENDPOINT
+              value: unix://plugin/csi.sock
+            - name: REST_URL
+              value: http://glusterd2-client.gcs:24007
```

Comment:
Whats this url is all about ? Is it a constant value for any GD2 cluster?

Reply:
It is just like the url in glusterfs-csi deployment file [1]. If it's deployed manually, this has to be changed which will be mentioned in README. If the deployment is part of gcs, gcs will take care of updating this.


[1] https://github.com/gluster/gluster-csi-driver/blob/28876cf008865e2ec365c7b61c34ccc30ab7411f/examples/kubernetes/csi-deployment.yaml#L153